com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ModuleHooksAggregator.java

```diff
@@ -0,0 +1,125 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections4.ListUtils;
+import org.flowable.engine.delegate.DelegateExecution;
+
+import com.sap.cloud.lm.sl.cf.core.model.HookPhase;
+import com.sap.cloud.lm.sl.cf.process.steps.StepsUtil;
+import com.sap.cloud.lm.sl.mta.model.Hook;
+import com.sap.cloud.lm.sl.mta.model.Module;
+
+public class ModuleHooksAggregator {
+
+    private static final byte MAJOR_SCHEMA_VERSION_THREE = 3;
+
+    private final DelegateExecution execution;
+    private final Module moduleToDeploy;
+
+    public ModuleHooksAggregator(DelegateExecution execution, Module moduleToDeploy) {
+        this.execution = execution;
+        this.moduleToDeploy = moduleToDeploy;
+    }
+
+    public List<Hook> aggregateHooks(HookPhase currentHookPhaseForExecution) {
+        substituteBeforeStartHooksPhases();
+        Map<String, List<String>> alreadyExecutedHooksForModule = getAlreadyExecutedHooks();
+        List<Hook> hooksCalculatedForExecution = determineHooksForExecution(alreadyExecutedHooksForModule, currentHookPhaseForExecution);
+        updateExecutedHooksForModule(alreadyExecutedHooksForModule, currentHookPhaseForExecution, hooksCalculatedForExecution);
+        return hooksCalculatedForExecution;
+    }
+
+    private void substituteBeforeStartHooksPhases() {
```

Comment:
I am concerned about this. First, why is this here? Second, this brings notation for the before start hook phases which is not correct from the Object point of view. It's purpose is to get some hooks, calculate some hooks for execution and return them.

Reply:
Removed. RestartAppStep will return the required phases